src/patch_parse.c

```diff
@@ -109,7 +109,10 @@ static int header_path_len(git_patch_parse_ctx *ctx)
 	size_t len;
 
 	for (len = quoted; len < ctx->line_len; len++) {
-		if (!quoted && git__isspace(ctx->line[len]))
+		if (!quoted &&
+		    (!git__strncmp(ctx->line + len, " b/", strlen(" b/")) ||
+		     !git__strncmp(ctx->line + len, " \"b/", strlen(" \"b/")) ||
```

Comment:
There's no guarantee that the prefixes are `a/` and `b/`.  You can set custom prefixes with `--src-prefix` and `--dst-prefix`, or you can have no prefix at all (with `--no-prefix`)...

This is a bit of a bummer as it means parsing the header line is completely ambiguous.  What does:

```
diff --git a/b c/d/e f/g h/i j/k
```

mean?

I think we'll need to store the whole header buffer then parse it with the knowledge given to us by the `---` and `+++` lines.

:(